source/linux/epoll_event_loop.c

```diff
@@ -562,6 +562,14 @@ static int aws_event_loop_listen_for_io_events(int epoll_fd, struct epoll_event
     return epoll_wait(epoll_fd, events, MAX_EVENTS, timeout);
 }
 
+static void s_aws_cleanup_aws_lc_thread_local_state(void *user_data) {
+    (void)user_data;
+
+#if defined(OPENSSL_IS_AWSLC)
+    AWSLC_thread_local_clear();
```

Comment:
Does aws-c-io always know where libcrypto's header files are? is aws-c-io making any other libcrypto calls directly?

Should there be some hideous wrapper in aws_cal_thread_cleanup()?

Or could we modify s2n so `s2n_cleanup()` calls this? Since their library already has the concept of "thread-cleanup" while ours do not have that concept so far

Reply:
I would assume so; aws-c-io has a direct link against against aws-c-cal

I don't think trying to push this into s2n_cleanup would be proper, especially since an event loop thread could conceivably invoke arbitrary aws-lc logic without ever involving s2n (consider a plaintext connection where crypto primitives were invoked in callbacks or the like);

Reply:
Nm, see what you mean now, although I'm disturbed as to why this worked as is.  Checking the addition of `#include <openssl/base.h>` now